core/src/test/java/io/parsingdata/metal/EqualityTest.java

```diff
@@ -196,24 +196,28 @@ public void parseGraph() {
 
     @Test
     public void stringRef() {
-        final Ref object = new NameRef("name");
-        assertFalse(object.equals(null));
+        final Ref<String> object = new NameRef("name");
+        assertNotEquals(null, object);
         assertNotEquals("name", object);
         assertEquals(object, new NameRef("name"));
+        assertNotEquals(object, new NameRef("name", "name"));
+        assertNotEquals(object, new NameRef("name", "otherName"));
         assertNotEquals(object, new NameRef("otherName"));
         assertNotEquals(object, new DefinitionRef(any("name")));
-        assertNotEquals(object, new NameRef("name", con(1)));
+        assertNotEquals(object, new NameRef( con(1), "name"));
```

Comment:
Swapped the arguments here to avoid using deprecated methods in our tests.

Reply:
Also remove the trailing white space:

```java
assertNotEquals(object, new NameRef( con(1), "name"));
                                    ^
```